x-pack/plugins/actions/server/create_execute_function.ts

```diff
@@ -46,12 +46,18 @@ export function createExecutionEnqueuerFunction({
       );
     }
 
-    const actionTypeId = await getActionTypeId(
+    const { actionTypeId, name, isMissingSecrets } = await getAction(
       unsecuredSavedObjectsClient,
       preconfiguredActions,
       id
     );
 
+    if (isMissingSecrets) {
+      throw new Error(
+        `Unable to execute action because the referenced connector is incorrectly configured. Please review the configuration for the "${name}" connector.`
```

Comment:
I think we might say here more concrete - about missing secrets. @gchaps what do you think?

Reply:
I recommend removing "An error occurred for unknown reasons", because it doesn't add any value, and you do, in fact, know the reason.  Here are two suggestions:

**One line** 

> Unable to execute action because no secrets are defined for the email connector.

**Title with description**

> Unable to execute action
> The referenced connector requires secrets.  Check the configuration for the email connector.

Reply:
Updated!